examples/petset/mongodb-petset.yaml

```diff
@@ -0,0 +1,147 @@
+kind: Template
+apiVersion: v1
+metadata:
+  name: mongodb-petset-replication
+  annotations:
+    description: "MongoDB Replication Example (based on PetSet). You must have persistent volumes available in your cluster to use this template."
+    iconClass: "icon-mongodb"
+    tags: "database,mongodb,replication"
+parameters:
+  - name: MONGODB_USER
+    displayName: "MongoDB Connection Username"
+    description: "Username for MongoDB user that will be used for accessing the database."
+    generate: expression
+    from: "[a-zA-Z0-9]{3}"
+    required: true
+
+  - name: MONGODB_PASSWORD
+    displayName: "MongoDB Connection Password"
+    description: "Password for the MongoDB connection user."
+    generate: expression
+    from: "[a-zA-Z0-9]{16}"
+    required: true
+
+  - name: MONGODB_DATABASE
+    displayName: "MongoDB Database Name"
+    description: "Name of the MongoDB database accessed."
+    value: sampledb
+    required: true
+
+  - name: MONGODB_ADMIN_PASSWORD
+    displayName: "MongoDB Admin Password"
+    description: "Password for the database admin user."
+    generate: expression
+    from: "[a-zA-Z0-9]{16}"
+    required: true
+
+  - name: MONGODB_REPLICA_NAME
+    displayName: "Replica Set Name"
+    description: "The name of the replica set."
+    value: rs0
+    required: true
+
+  - name: MONGODB_KEYFILE_VALUE
+    displayName: "Keyfile Content"
+    description: "The value of the MongoDB keyfile (https://docs.mongodb.com/manual/core/security-internal-authentication/#internal-auth-keyfile)."
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
+    required: true
+
+  - name: MONGODB_IMAGE
+    displayName: "MongoDB Docker Image"
+    description: "A reference to a supported MongoDB Docker image (centos/mongodb-32-centos7, centos/mongodb-26-centos7, openshift/mongodb-24-centos7, ...)."
+    value: "openshift/mongodb-24-centos7"
```

Comment:
I think we should default to the 3.2 image before we merge.

@php-coder how about instead of using 2.4 we switch to use the 3.2 image as the "working image" (and keep new files only there to ease development), then backport to 2.4 and 2.6?